pkg/stanza/adapter/config.go

```diff
@@ -18,24 +18,18 @@ import (
 	"time"
 
 	"go.opentelemetry.io/collector/config"
-	"gopkg.in/yaml.v2"
 
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
 )
 
 // BaseConfig is the common configuration of a stanza-based receiver
 type BaseConfig struct {
 	config.ReceiverSettings `mapstructure:",squash"`
-	Operators               OperatorConfigs     `mapstructure:"operators"`
```

Comment:
this changes the type. I am confused.

Reply:
This PR is a superset of https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/14174, so some extra changes showing in the diff here.